src/frequenz/channels/experimental/_grouping_latest_value_cache.py

```diff
@@ -220,6 +231,50 @@ def __delitem__(self, key: HashableT) -> None:
         """
         del self._latest_value_by_key[key]
 
+    @typing.overload
+    def pop(self, key: HashableT, /) -> ValueT_co | None:
+        """Remove the latest value for a specific key and return it."""
+
+    @typing.overload
+    def pop(self, key: HashableT, /, default: DefaultT) -> ValueT_co | DefaultT:
+        """Remove the latest value for a specific key and return it."""
+
+    def pop(
+        self, key: HashableT, /, default: DefaultT | _NotSpecified = _NotSpecified()
+    ) -> ValueT_co | DefaultT | None:
```

Comment:
```suggestion
    def pop(
        self, key: HashableT, /, default: DefaultT | None = None
    ) -> ValueT_co | DefaultT | None:
```

Then you can use `return self._latest_value_by_key.pop(key, default)` as the only body, because that's the exact signature of dict.pop(), and no need for special cases.

And if a user wants to have `None` in `ValueT_co`, they can make their own sentinel that will be `DefaultT`.  So we don't need to make a sentinel to be able to satisfy `MutableMapping`.

Reply:
No, the problem is `pop()` raises when the `default` is not specified, it doesn't return `None`, so we need to be able to tell between not passing the argument and passing `None`.

Reply:
oh damn, that's such a bad API.

Reply:
![image](https://github.com/user-attachments/assets/24b498d0-aeac-48e5-862d-fa7190af05f8)